tests/e2e/test_api.py

```diff
@@ -89,6 +89,7 @@ def teardown_module(module):
     """Clean up the environment after all tests are executed."""
     if on_cluster:
         must_gather()
+        cluster_utils.delete_user("test-user")
```

Comment:
if you're going to cleanup the things that were done in setup, there's more than this to address (there is also the metrics-test-user, and some rolebindings).

it doesn't generally matter since they just get reused as is so it's ok if they don't get cleaned up, but it would be better if we were thorough lest someone see this and think there's a specific reason we are cleaning up some objects but not others.